src/pydrex/minerals.py

```diff
@@ -98,185 +100,155 @@ class Mineral:
+                dislocation_exponent=config["dislocation_exponent"],
+                nucleation_efficiency=config["nucleation_efficiency"],
+                gbm_mobility=config["gbm_mobility"],
+                volume_fraction=volume_fraction,
+            )
+            return np.hstack((
+                np.dot(velocity_gradient, finite_strain_ell).flatten(),
+                rotation_rates.flatten() * strain_rate_max,
+                fractions_diff * strain_rate_max
+            ))
+
+        # Imposed macroscopic strain rate tensor.
+        strain_rate = (velocity_gradient + velocity_gradient.transpose()) / 2
+        # Strain rate scale (max. eigenvalue of strain rate).
+        strain_rate_max = np.abs(la.eigvalsh(strain_rate)).max()
+        # Dimensionless strain rate and velocity gradient.
+        nondim_strain_rate = strain_rate / strain_rate_max
+        nondim_velocity_gradient = velocity_gradient / strain_rate_max
+        # Volume fraction of the mineral phase.
         if self.phase == MineralPhase.olivine:
             volume_fraction = config["olivine_fraction"]
         elif self.phase == MineralPhase.enstatite:
             volume_fraction = config["enstatite_fraction"]
         else:
             assert False  # Should never happen.
 
-        # ========== RK step  1 ==========
-        orientations_diff, fractions_diff = _core.derivatives(
-            phase=self.phase,
-            fabric=self.fabric,
-            n_grains=self.n_grains,
-            orientations=self.orientations,
-            fractions=self.fractions,
-            strain_rate=strain_rate,
-            velocity_gradient=velocity_gradient,
-            stress_exponent=stress_exponent,
-            dislocation_exponent=dislocation_exponent,
-            nucleation_efficiency=nucleation_efficiency,
-            gbm_mobility=gbm_mobility,
-            volume_fraction=volume_fraction,
-        )
-        orientations_1 = orientations_diff * dt * strain_rate_max
-        orientations_iter = self.orientations + 0.5 * orientations_1
-        orientations_iter.clip(-1, 1)
-        fractions_1 = fractions_diff * dt * strain_rate_max
-        fractions_iter = self.fractions + 0.5 * fractions_1
-        fractions_iter.clip(0, None)
-        fractions_iter /= fractions_iter.sum()
-
-        # ========== RK step  2 ==========
-        orientations_diff, fractions_diff = _core.derivatives(
-            phase=self.phase,
-            fabric=self.fabric,
-            n_grains=self.n_grains,
-            orientations=self.orientations,
-            fractions=self.fractions,
-            strain_rate=strain_rate,
-            velocity_gradient=velocity_gradient,
-            stress_exponent=stress_exponent,
-            dislocation_exponent=dislocation_exponent,
-            nucleation_efficiency=nucleation_efficiency,
-            gbm_mobility=gbm_mobility,
-            volume_fraction=volume_fraction,
+        sol = RK45(
```

Comment:
Small comment here, we are using RK45 at the moment, but we may end up changing to another solver depending on future benchmarks/models.